helper/jsonutil/json.go

```diff
@@ -46,3 +74,39 @@ func DecodeJSONFromReader(r io.Reader, out interface{}) error {
 	// Since 'out' is an interface representing a pointer, pass it to the decoder without an '&'
 	return dec.Decode(out)
 }
+
+// DecompressAndDecodeJSON tries to decompress the given data. The call to
+// decompress, fails if the content was not compressed in the first place,
+// which is identified by a canary byte before the compressed data. If the data
+// is not compressed, it is JSON decoded directly. Otherwise the decompressed
+// data will be JSON decoded.
+func DecompressAndDecodeJSON(dataBytes []byte, out interface{}) error {
```

Comment:
We shouldn't need this function. The normal DecodeJSON should check if the first byte matches one of the canaries and do the right thing.